mantle/cmd/kola/switchkernel.go

```diff
@@ -0,0 +1,249 @@
+	cmdSwitchKernel.MarkFlagRequired("kernel-rt")
+	root.AddCommand(cmdSwitchKernel)
+}
+
+func runSwitchKernel(cmd *cobra.Command, args []string) error {
+	// currently only supports RHCOS
+	if kola.Options.Distribution != "rhcos" {
+		return fmt.Errorf("Only supports `rhcos`")
+	}
+
+	var userdata *conf.UserData = conf.Ignition(fmt.Sprintf(`{
+		"ignition": {
+			"version": "2.2.0"
+		},
+		"storage": {
+			"files": [
+				{
+					"filesystem": "root",
+					"path": "/var/home/core/switch-kernel.sh",
+					"contents": {
+						"source": "data:text/plain;base64,%s"
+					},
+					"mode": 110
+				}
+			]
+		}
+	}`, base64.StdEncoding.EncodeToString([]byte(switchKernelScript))))
+
+	flight, err := kola.NewFlight(kolaPlatform)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create new flight")
+	}
+	defer flight.Destroy()
+
+	c, err := flight.NewCluster(&platform.RuntimeConfig{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create new cluster")
+	}
+	defer c.Destroy()
+
+	m, err := c.NewMachine(userdata)
+	if err != nil {
+		return errors.Wrapf(err, "failed to spawn new machine")
+	}
+	defer m.Destroy()
+
+	err = testSwitchKernel(c)
+	if err != nil {
+		return errors.Wrapf(err, "failed switch kernel test")
+	}
+
+	return nil
+}
+
+// Drops Kernel RT RPM files under the directory `localPath` to `$homeDir/kernel-rt-rpms` directory in m
+func dropRpmFilesAll(m platform.Machine, localPath string) error {
+	fmt.Println("Dropping RT Kernel RPMs...")
+	re := regexp.MustCompile(`^kernel-rt-.*\.rpm$`)
+	files, err := ioutil.ReadDir(localPath)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		filename := f.Name()
+		filepath := strings.TrimSuffix(localPath, "/") + "/" + filename
+		targetPath := fmt.Sprintf("%s/kernel-rt-rpms/%s", homeDir, filename)
+		if re.MatchString(filename) {
+			in, err := os.Open(filepath)
+			if err != nil {
+				return err
+			}
+			if err := platform.InstallFile(in, m, targetPath); err != nil {
+				return errors.Wrapf(err, "failed to upload %s", filename)
+			}
+		}
+	}
+	return nil
+}
+
+func switchDefaultToRtKernel(c platform.Cluster, m platform.Machine) error {
+	// run the script to switch from default kernel to rt kernel
+	fmt.Println("Switching from Default to RT Kernel...")
+	cmd := "sudo " + homeDir + "/switch-kernel.sh default rt-kernel " + homeDir + "/kernel-rt-rpms/"
+	stdout, stderr, err := m.SSH(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run %s", cmd)
+	}
+	fmt.Println(fmt.Sprintf("%s", stderr))
+	fmt.Println(fmt.Sprintf("%s", stdout))
+
+	// reboot the machine to switch kernel
+	fmt.Println("Rebooting machine...")
+	err = m.Reboot()
+	if err != nil {
+		return errors.Wrapf(err, "failed to reboot machine")
+	}
+
+	// check if the kernel has switched to rt kernel
+	fmt.Println("Checking kernel type...")
+	cmd = "uname -v | grep -q 'PREEMPT RT'"
+	_, _, err = m.SSH(cmd)
+	if err == nil {
```

Comment:
This should be `err != nil` right?

Hmm...I think this test is passing because you're missing a `_` in `PREEMPT_RT` so both tests are inverted?

Reply:
Double checked inside a rhcos machine..
```
[core@ibm-p8-kvm-03-guest-02 ~]$ uname -v 
#1 SMP PREEMPT RT Tue Jan 14 16:03:46 UTC 2020
```

Reply:
Ah OK.  But then I think we should invert these checks, because `if err == nil` is a HUGE trap in Go, it just looks like a typo.

And specifically here the `error.Wrapf()` is wrong for this one because `err` is nil, so it will return `nil`, and the rest of the code will think no error occurred.

Reply:
The `make check` ran by `openshift-ci-rebot`  didn't run `switchkernel.go` at all, it only ran `kola --help` for sanity check.

Yes, the it should be `if err != nil`, sorry should've mentioned earlier.. and I confirmed on RHCOS the original test did not produce expected result